dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/generate/ServiceImplementationGenerator.java

```diff
@@ -393,4 +394,51 @@ public CodeBlock multimapParam() {
             }
         });
     }
+
+    private CodeBlock generateListElementSerializerCall(CodeBlock argName, ArgumentType type) {
+        CodeBlock elementName = CodeBlock.of("$L$L", argName, "Element");
+        return type.match(new ArgumentType.Cases<>() {
+            @Override
+            public CodeBlock primitive(TypeName _typeName, String parameterSerializerMethodName) {
+                return CodeBlock.of(
+                        ".$L($L -> $L.$L($L))",
+                        MAP,
+                        elementName,
+                        PARAMETER_SERIALIZER,
+                        parameterSerializerMethodName,
+                        elementName);
+            }
+
+            @Override
+            public CodeBlock list(TypeName _typeName, ListType _listType) {
+                throw new UnsupportedOperationException("This should not happen");
+            }
+
+            @Override
+            public CodeBlock alias(TypeName _typeName, String parameterSerializerMethodName) {
+                return CodeBlock.of(
+                        ".$L($L -> $L.$L($L.get()))",
+                        MAP,
+                        elementName,
+                        PARAMETER_SERIALIZER,
+                        parameterSerializerMethodName,
+                        elementName);
```

Comment:
Thoughts on splitting this into two maps on method references instead of a single complex lambda? I thought that's what @pkoenig10 was asking in https://github.com/palantir/dialogue/pull/2079#discussion_r1424348673

`.map(AliasType::get).map(_parameterSerializer::parameterSerializerMethodName)`

Reply:
Fixed